modules/ROOT/pages/patterns/match-modes.adoc

```diff
@@ -0,0 +1,254 @@
+----
+
+<1> The xref:expressions/list-expressions.adoc#list-comprehension[list comprehensions] iterate over nodes and relationships in a path and return specific properties (the `name` property from nodes, and the `id` property from relationships).
+
+As the results show, a node may be traversed several times in the same path, but a relationship (bridge) is never re-traversed (for example, the path `[7, 2, 5, 7, 2]`, which traverses the same bridge more than once, cannot be matched using the `DIFFERENT RELATIONSHIPS` match mode).
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| locations | crossedBridges 
+
+| ["Kneiphof", "North Bank", "Kneiphof", "South Bank", "Kneiphof", "Lomse"]
+| [1, 5, 6, 4, 7]
+
+| ["Kneiphof", "South Bank", "Kneiphof", "North Bank", "Kneiphof", "Lomse"] 
+| [6, 4, 1, 5, 7]
+
+2+d|Rows: 2
+|===
+
+In the next example, the direction is removed from the pattern, and the path length is increased to `6`.
+It shows that there are `48` valid paths with this length, where no relationships (bridges) are re-traversed, and the query returns one of those paths as a `samplePath`, showing both the nodes and relationships traversed therein.
+
+.Find paths with a length of 6 relationships from a start node
+[source, cypher]
+----
+MATCH p = (:Location {name: 'Kneiphof'})-[:BRIDGE]-{6}()
+WITH count(p) AS pathCount, collect(p)[0] AS samplePath // <1>
+ORDER BY pathCount
+RETURN pathCount, 
+       [n IN nodes(samplePath) | n.name] AS samplePathLocations,
+       [r IN relationships(samplePath) | r.id] AS samplePathBridges
+----
+
+<1> The xref:functions/list.adoc#functions-collect[`collect()`] function collects all paths and `[0]` takes the first entry as the `samplePath`.
+
+.Result
+[role="queryresult",options="header,footer",cols="3*<m"]
+|===
+| pathCount | samplePathLocations | samplePathBridges
+
+| 48
+| ["Kneiphof", "North Bank", "Kneiphof", "South Bank", "Kneiphof", "Lomse", "North Bank"]
+| [1, 5, 6, 4, 7, 2]
+
+3+d|Rows: 1
+|===
+
+image::different-relationships-graph.svg[Graph showing all but one of the seven bridges of Königsberg traversed, width=500, role=popup]
+_In the `samplePath` with a length of `6` returned by `DIFFERENT RELATIONSHIPS`, all bridges are traversed except for bridge `3`._
+
+However, if the relationship count is increased to `7`, `0` paths are returned.
+
+.Find paths with a length of 7 relationships
+[source, cypher]
+----
+MATCH p = (:Location {name: 'Kneiphof'})--{7}()
+RETURN count(p) AS pathCount
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="1*<m"]
+|===
+| pathCount
+
+| 0
+
+1+d|Rows: 1
+|===
+
+In the `DIFFERENT RELATIONSHIPS` match mode, each step must use a unique bridge.
+After the first step, one bridge is used, leaving six remaining bridges to be used in the subsequent steps.
+Once six remaining bridges are traversed, a seventh step would require re-traversing a bridge, which is not allowed. 
```

Comment:
This is still incorrect. Crossing the six remaining bridges (after the first mentioned in the previous sentence) the seventh would have been crossed. Either say "once five more bridges are traversed" or "after the sixth" or something equivalent.

Reply:
Ah you fixed it. Great.